blaze-client/src/main/scala/org/http4s/blaze/client/BlazeClient.scala

```diff
@@ -90,20 +92,42 @@ private class BlazeClient[F[_], A <: BlazeConnection[F]](
 )(implicit F: ConcurrentEffect[F])
     extends DefaultClient[F] {
 
+  private[this] val logger = getLogger
+
   override def run(req: Request[F]): Resource[F, Response[F]] =
-    runLoop(req, retries)
+    Hotswap.create[F, Either[Throwable, Response[F]]].flatMap { hotswap =>
+      Resource.eval(retryLoop(req, retries, hotswap))
+    }
 
-  private def runLoop(req: Request[F], remaining: Int): Resource[F, Response[F]] = {
+  // A light implementation of the Retry middleware.  That needs a
+  // Timer, which we don't have.
```

Comment:
In my desperation to get the tests passing, I found fs2 had an implementation, so this actually looks more like 0.23's Retry, and we could rebuild 0.22's on it.  It's cleaner, but I don't want to touch something on an aging branch not known to have bugs.